app/controllers/jobs_controller.rb

```diff
@@ -0,0 +1,58 @@
+class JobsController < ApplicationController
+  before_action :get_company
+  before_action :set_job, only: %i[show edit update destroy]
+
+  after_action :verify_authorized, except: %i[index show]
+
+  def index; end
+
+  def show; end
+
+  def new
+    @job = @company.jobs.new
+    authorize @job
+  end
+
+  def edit
+    authorize @job
+  end
+
+  def create
+    @job = @company.jobs.new(job_params)
+    authorize @job
+    if @job.save
+      redirect_to companies_path(@company), notice: 'Job was successfully created.'
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    authorize @job
+    if @job.update(job_params)
+      redirect_to company_path(@company), notice: 'Job was successfully updated.'
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    authorize @job
+    @job.destroy
```

Comment:
if destroy, redirect. Otherwise, redirect back to the index page saying that it could not be deleted and to try again. 

There could be validations that we could add that would prevent a record to be deleted similar to other validations. (We don't have this now, but we could later and our app needs to be able to support that).